### DISCUSSION TITLE
How often do get_variable and get_config (aka secrets backend) get called?

### DISCUSSION THREAD

**Discussion by houssam7737** [apache/airflow#26935] [Q&A] [answered]
Hi All,

I'm dealing with a Vault outage because of a high number of requests coming from Airflow instances. I'm trying to understand how many times secrets backend methods get called, and failing to find hints in the docs. I also understand that this is highly dependent on configuration, but what configs does it depend on? Are secrets fetched at every dag refresh? Every worker refresh? I also have a git-sync sidecar which I'm sure is responsible for some of those requests.

Any help would be much appreciated! Thanks a lot!

- Houssam

**Comment by potiuk** [comment, marked as answer]
If you are using variables incorrectly - in top-level code then yes, it will be called with every dag parse (every 30 s. by default). You should follow the Best practices and avoid using any similar operation in it:  https://airflow.apache.org/docs/apache-airflow/stable/best-practices.html?highlight=best+practices#top-level-python-code

Git Sync has nothing to do with it most likely. It's continuous parsing of the DAG files that is the reason. 

Move your variable reading to JINJA templates or to inside "execute" of your operators or to inside of @task decorated functions.

**Reply by houssam7737** [reply]
This was exactly it. Thank you!!!
